packages/addons-v5/commands/addons/wait.js

```diff
@@ -1,48 +1,59 @@
+/* eslint-disable no-await-in-loop */
+
 'use strict'
 
 const cli = require('heroku-cli-util')
 const {notify} = require('../../lib/notify')
-const waitForAddonProvisioning = require('../../lib/addons_wait')
+const {waitForAddonProvisioning, waitForAddonDeprovisioning} = require('../../lib/addons_wait')
 
 async function run(ctx, api) {
   const resolve = require('../../lib/resolve')
 
   let addons
   if (ctx.args.addon) {
     addons = [await resolve.addon(api, ctx.app, ctx.args.addon)]
+  } else if (ctx.app) {
+    addons = await api.get(`/apps/${ctx.app}/addons`)
   } else {
-    if (ctx.app) { // eslint-disable-line no-lonely-if
-      addons = await api.get(`/apps/${ctx.app}/addons`)
-    } else {
-      addons = await api.get('/addons')
-    }
+    addons = await api.get('/addons')
   }
 
-  addons = addons.filter(addon => addon.state === 'provisioning')
+  addons = addons.filter(addon => addon.state === 'provisioning' || addon.state === 'deprovisioning')
 
-  let interval = Number.parseInt(ctx.flags['wait-interval'])
+  let interval = Number.parseInt(ctx.flags['wait-interval'], 10)
   if (!interval || interval < 0) {
     interval = 5
   }
 
   for (let addon of addons) {
     const startTime = new Date()
-    try {
-      addon = await waitForAddonProvisioning(api, addon, interval)
-    } catch (error) {
-      notify(`heroku addons:wait ${addon.name}`, 'Add-on failed to provision', false)
-      throw error
-    }
 
-    let configVars = (addon.config_vars || [])
-    if (configVars.length > 0) {
-      configVars = configVars.map(c => cli.color.configVar(c)).join(', ')
-      cli.log(`Created ${cli.color.addon(addon.name)} as ${configVars}`)
-    }
+    if (addon.state === 'provisioning') {
+      try {
+        addon = await waitForAddonProvisioning(api, addon, interval)
+      } catch (error) {
+        notify(`heroku addons:wait ${addon.name}`, 'Add-on failed to provision', false)
+        throw error
+      }
+
+      let configVars = (addon.config_vars || [])
+
+      if (configVars.length > 0) {
+        configVars = configVars.map(c => cli.color.configVar(c)).join(', ')
+        cli.log(`Created ${cli.color.addon(addon.name)} as ${configVars}`)
+      }
+
+      // only show notification if addon took longer than 5 seconds to provision
+      if (Date.now() - startTime >= 1000 * 5) {
+        notify(`heroku addons:wait ${addon.name}`, 'Add-on successfully provisioned')
+      }
+    } else if (addon.state === 'deprovisioning') {
+      addon = await waitForAddonDeprovisioning(api, addon, interval)
```

Comment:
Please create a new var instead of overwriting `addon`.

Reply:
I can do that, but it's the exact same add-on and I was just going with the flow of how it was implemented the async provisioning part. Should I take this approach with the rest of the code to keep consistency?

Reply:
I'm removing that particular assignment on that line, because the return value isn't used at all, but in other parts of the same file the `addon` var gets reused, so I'm changing those lines to keep consistency.